### ISSUE TITLE
Newline error

### ISSUE THREAD

**Issue by CMCDragonkai** [iwonbigbro/bashpp#13]
This simple file fails to be parsed:

```
#ifdef LOL

hahahah

#endif
```

It requires an empty newline:

```
#ifdef LOL

hahahah

#endif

```

The error is:

```
 ± ./bpp -D LOL testfile                                                                                                                    16-03-31 22:48:16

hahahah

Traceback:
  ./bpp:iter_file():1590:         die "frozen buffer"
  ./bpp:iter_files():1517:         iter_file "$f"
  ./bpp:main():1761: iter_files "${files[@]}" >"$output"
testfile:5:7 error: frozen buffer
  #endif
  ......^
```

By the way, this is freaking awesome!

**Comment by iwonbigbro** [comment]
Nice find.  I'll see if I can get it fixed soon.

**Comment by CMCDragonkai** [comment]
The bug affects files included via `#include` as well. Any file included require an empty newline or we get an error.

**Comment by CMCDragonkai** [comment]
@iwonbigbro Any news on this?

**Comment by CMCDragonkai** [comment]
Basically we need to do extra processing for when we hit the end of file. 

I found this that this part is the relevant area: https://github.com/iwonbigbro/bashpp/blob/master/bin/bashpp#L1566-L1585

Basically we need to add in a part to deal for when an end of file is met.

```
    if (( getc_eof )) ; then
        # do something here
    fi
```

However I'm not sure what to add there to pop the 2 states from the stack.

**Comment by iwonbigbro** [comment]
Sorry, been super busy.  Lemme take a look.

**Comment by iwonbigbro** [comment]
Oh yeah, so basically, `process_char` is processing the next character along with what we have queued up in the character pipeline thus far (`pr_c`).  So all we need is an additional call to `process_char "$c" "$pr_c"`.  However, we can't just call that after the loop, because we'd call it twice on the same buffer for files that do terminate with a `\n`.  What we need is a little bit of re-organising of the `getc` call, so we can ensure we always call `process_char` if there is something in the buffer or we got a new character.
